mvcc/index_bench_test.go

```diff
@@ -0,0 +1,25 @@
+package mvcc
+
+import (
+	"testing"
+)
+
+func BenchmarkIndexCompact1(b *testing.B)   { benchmarkIndexCompact(b, 1) }
+func BenchmarkIndexCompact100(b *testing.B) { benchmarkIndexCompact(b, 100) }
+func BenchmarkIndexCompact10000(b *testing.B) { benchmarkIndexCompact(b, 10000) }
+func BenchmarkIndexCompact100000(b *testing.B) { benchmarkIndexCompact(b, 100000) }
+func BenchmarkIndexCompact1000000(b *testing.B) { benchmarkIndexCompact(b, 1000000) }
+
+func benchmarkIndexCompact(b *testing.B, size int) {
```

Comment:
do you have a result for the benchmark with a comparison with the result before the patch?

Reply:
On local development machine -

master:
```
go test github.com/coreos/etcd/mvcc -v -run=^$ -bench BenchmarkIndexCompact
goos: linux
goarch: amd64
pkg: github.com/coreos/etcd/mvcc
BenchmarkIndexCompact1-16                1000000              1655 ns/op
BenchmarkIndexCompact100-16               100000             19772 ns/op
BenchmarkIndexCompact10000-16               2000            880883 ns/op
BenchmarkIndexCompact100000-16               200           8725995 ns/op
BenchmarkIndexCompact1000000-16              100         386072323 ns/op
PASS
```

this branch:
```
go test github.com/coreos/etcd/mvcc -v -run=^$ -bench BenchmarkIndexCompact  
goos: linux
goarch: amd64
pkg: github.com/coreos/etcd/mvcc
BenchmarkIndexCompact1-16                1000000              1572 ns/op
BenchmarkIndexCompact100-16               100000             22216 ns/op
BenchmarkIndexCompact10000-16               2000           1058001 ns/op
BenchmarkIndexCompact100000-16               100          10484775 ns/op
BenchmarkIndexCompact1000000-16              100         423129293 ns/op
```

Added time likely due to the additional lock/unlock calls.